source/Registry.cpp

```diff
@@ -176,13 +200,25 @@ void Registry::join_with(const FieldModel& field_model) {
   }
 }
 
+void Registry::join_with(const LiteralModel& literal_model) {
+  auto iterator = literal_models_.find(literal_model.pattern());
+  if (iterator != end(literal_models_)) {
```

Comment:
nitpick: we usually use `literal_models_.end()`. Let's be consistent with the rest of the codebase.